core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -250,11 +254,7 @@ public void writeTo(StreamOutput out) throws IOException {
         boolean hasIndexBoost = indexBoost != null;
         out.writeBoolean(hasIndexBoost);
```

Comment:
While we are able to break the wire protocol, can you remove the boolean and just use size=0 on read to avoid building the map? May as well size a boolean over the wire and in the cache key